ktor-http/common/src/io/ktor/http/content/OutgoingContent.kt

```diff
@@ -56,6 +56,11 @@ public sealed class OutgoingContent {
         }
     }
 
+    /**
+     * Trailers to set when sending this content, will be ignored if request is not in HTTP2 mode
```

Comment:
Btw, trailing headers are possible in HTTP/1.1 but usually not processed and almost nobody cares. The question is: should we try to support it in HTTP/1 or we state that it is HTTP/2 only.

The other issue is that it's not easy to use since we have many content implementations and helper functions most of them don't provide a way to specify even regular headers. So how one could provide these headers?

Reply:
For HTTP/1.1, I think we should leave it right now since there is no use case for it.

GRPC only works on HTTP2. It utilizes the flow control for back pressure (I guess here it's already handled by Ktor so no need to worry about it), it also uses trailer to send its own status code, so the status can be sent after the stream.

This is a rarely used feature (seems except for GRPC no one uses it), it should be available on the most flexible API (OutgoingContent I assume), but not others for simplicity.